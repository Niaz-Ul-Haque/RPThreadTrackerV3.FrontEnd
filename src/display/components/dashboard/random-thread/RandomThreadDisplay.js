import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	thread: PropTypes.shape({}).isRequired
};

const RandomThreadDisplay = (props) => {
	const { thread } = props;
	if (!thread.userTitle) {
		return (
			<div className="random-thread-result">
				<p>Pick a random thread to respond to!</p>
			</div>
		);
	}
	return (
		<div className="random-thread-result">
			<div>
				<a href={thread.lastPostUrl}>
					{thread.userTitle}
					<i className="fa fa-external-link" />
				</a>
			</div>
			<div className="small ">
				Last Post by <a href={thread.lastPostUrl}>{thread.lastPosterUrlIdentifier}</a>
			</div>
		</div>
	);
};

RandomThreadDisplay.propTypes = propTypes;
export default RandomThreadDisplay;
