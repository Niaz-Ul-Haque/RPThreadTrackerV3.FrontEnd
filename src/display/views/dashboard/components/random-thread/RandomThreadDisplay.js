import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	threadData: PropTypes.shape({}).isRequired
};

const RandomThreadDisplay = (props) => {
	const { threadData } = props;
	if (!threadData.thread) {
		return (
			<div className="random-thread-result">
				<p>Pick a random thread to respond to!</p>
			</div>
		);
	}
	return (
		<div className="random-thread-result">
			<div>
				<a href={threadData.status.LastPostUrl}>
					{threadData.thread.userTitle}{' '}
					<i className="fa fa-external-link" />
				</a>
			</div>
			{threadData.status &&
				<div className="small ">
					Last Post by{' '}
					<a href={threadData.status.LastPostUrl}>{threadData.status.LastPosterUrlIdentifier}</a>
				</div>
			}
			{!threadData.status &&
				<div className="small ">
					Awaiting Starter
				</div>
			}
		</div>
	);
};

RandomThreadDisplay.propTypes = propTypes;
export default RandomThreadDisplay;
