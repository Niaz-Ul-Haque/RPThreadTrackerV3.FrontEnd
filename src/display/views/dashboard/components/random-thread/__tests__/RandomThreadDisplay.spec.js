// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import RandomThreadDisplay from '../RandomThreadDisplay';
// #endregion imports

const getValidProps = () => ({
	thread: {
		userTitle: 'Test Title'
	},
	status: {
		lastPostUrl: 'testurl',
		lastPosterUrlIdentifier: 'demouser'
	}
});
const getNullStatusProps = () => ({
	thread: {
		userTitle: 'Test Title'
	},
	status: null
});
const getNullThreadProps = () => ({
	thread: null
});

describe('rendering', () => {
	describe('snapshots', () => {
		it('should render valid snapshot when thread is null', () => {
			const element = shallow(<RandomThreadDisplay threadData={getNullThreadProps()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot without status', () => {
			const element = shallow(<RandomThreadDisplay threadData={getNullStatusProps()} />);
			expect(element).toMatchSnapshot();
		});
		it('should render valid snapshot with status', () => {
			const element = shallow(<RandomThreadDisplay threadData={getValidProps()} />);
			expect(element).toMatchSnapshot();
		});
	});
});
