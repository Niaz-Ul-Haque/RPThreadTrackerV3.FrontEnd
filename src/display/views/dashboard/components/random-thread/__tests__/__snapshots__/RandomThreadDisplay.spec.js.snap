// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot when thread is null 1`] = `
<div
  className="random-thread-result"
  data-spec="random-thread-result"
>
  <p>
    Pick a random thread to respond to!
  </p>
</div>
`;

exports[`rendering snapshots should render valid snapshot with status 1`] = `
<div
  className="random-thread-result"
  data-spec="random-thread-result"
>
  <div>
    <p
      data-spec="random-thread-title"
    >
      <a
        href="testurl"
      >
        Test Title
         
        <i
          className="fas fa-external-link-alt"
        />
      </a>
    </p>
    <div
      className="small"
      data-spec="random-thread-subtitle"
    >
      Last Post by
       
      <a
        href="testurl"
      >
        blackjackkent
      </a>
    </div>
  </div>
</div>
`;

exports[`rendering snapshots should render valid snapshot without status 1`] = `
<div
  className="random-thread-result"
  data-spec="random-thread-result"
>
  <div>
    <p
      data-spec="random-thread-title"
    >
      Test Title
    </p>
    <div
      className="small"
      data-spec="random-thread-subtitle"
    >
      Awaiting Starter
    </div>
  </div>
</div>
`;
